src/PermissionedUSDCWrapper.sol

```diff
@@ -0,0 +1,193 @@
+    }
+
+    /**
+     * @dev See {ERC20-decimals}.
+     */
+    function decimals() public view virtual override(ERC20, ERC20Wrapper) returns (uint8) {
+        try IERC20Metadata(address(_underlying)).decimals() returns (uint8 value) {
+            return value;
+        } catch {
+            return super.decimals();
+        }
+    }
+
+    /**
+     * @dev Allow a user to deposit underlying tokens and mint the corresponding number of wrapped tokens.
+     */
+    function depositFor(address account, uint256 value) public override returns (bool) {
+        address sender = _msgSender();
+        if (sender == address(this)) {
+            revert ERC20InvalidSender(address(this));
+        }
+        if (account == address(this)) {
+            revert ERC20InvalidReceiver(account);
+        }
+        SafeERC20.safeTransferFrom(_underlying, sender, address(this), value);
+        _mint(account, value);
+        return true;
+    }
+
+    /**
+     * @dev Allow a user to burn a number of wrapped tokens and withdraw the corresponding number of underlying tokens.
+     */
+    function withdrawTo(address account, uint256 value) public override returns (bool) {
+        if (account == address(this)) {
+            revert ERC20InvalidReceiver(account);
+        }
+        _burn(_msgSender(), value);
+        SafeERC20.safeTransfer(_underlying, account, value);
+        return true;
+    }
+
+    function hasPermission(address account) public view returns (bool attested) {
+        if (
+            account == address(this) || account == address(0) || account == MORPHO || account == BUNDLER
+                || memberlist.isMember(account)
+        ) {
+            return true;
+        }
+
+        Attestation memory verifiedAccountAttestation = getVerifiedAccountAttestation(account);
+        bool isAccountVerified = keccak256(verifiedAccountAttestation.data) == keccak256(abi.encodePacked(uint256(1)));
+
+        Attestation memory verifiedCountryAttestation = getVerifiedCountryAttestation(account);
+        string memory countryCode = parseCountryCode(verifiedCountryAttestation.data);
+        bool isUS = keccak256(abi.encodePacked(countryCode)) == keccak256(abi.encodePacked("US"));
+
+        return isAccountVerified && !isUS;
+    }
+
+    function getVerifiedCountryAttestation(address account) public view returns (Attestation memory attestation) {
+        bytes32 attestationUid = attestationIndexer.getAttestationUid(account, verifiedCountrySchemaUid);
+        require(attestationUid != 0, "USDCWrapper: no attestation found");
+        attestation = attestationService.getAttestation(attestationUid);
+        require(attestation.expirationTime == 0, "USDCWrapper: attestation expired");
+        require(attestation.revocationTime == 0, "USDCWrapper: attestation revoked");
+    }
+
+    function getVerifiedAccountAttestation(address account) public view returns (Attestation memory attestation) {
```

Comment:
seems like the code in `getVerifiedCountryAttestation` & `getVerifiedAccountAttestation` is doing the same thing. Would make sense to have one generic function where you only pass the schema id given the returned attestation struct is the same